protos/geofence/geofence.proto

```diff
@@ -26,20 +28,33 @@ message Point {
     double longitude_deg = 2; // Longitude in degrees (range: -180 to +180)
 }
 
+// Geofence types.
+enum FenceType {
+    FENCE_TYPE_INCLUSION = 0; // Type representing an inclusion fence
+    FENCE_TYPE_EXCLUSION = 1; // Type representing an exclusion fence
+}
+
 // Polygon type.
 message Polygon {
-    // Geofence polygon types.
-    enum FenceType {
-        FENCE_TYPE_INCLUSION = 0; // Type representing an inclusion fence
-        FENCE_TYPE_EXCLUSION = 1; // Type representing an exclusion fence
-    }
-
     repeated Point points = 1; // Points defining the polygon
     FenceType fence_type = 2; // Fence type
 }
 
-message UploadGeofenceRequest {
+// Circular type.
+message Circle {
+    Point point = 1; // Point defining the center
+    float radius = 2 [(mavsdk.options.default_value)="NaN"]; // Radius of the circular fence
+    FenceType fence_type = 3; // Fence type
+}
+
+// Geofence data type.
+message GeofenceData {
     repeated Polygon polygons = 1; // Polygon(s) representing the geofence(s)
+    repeated Circle circles = 2; // Circle(s) representing the geofence(s)
```

Comment:
@JonasVautherin that sort of extension should be ok and backwards compatible, right?

I guess anyway, this will go into MAVSDK 2.0.

Reply:
At the protobuf level, yes. But when autogenerating our API, I think it will break it (now the constructor will take two objects instead of one).

Reply:
I think that's ok for 2.0 then.